string.c

```diff
@@ -10963,9 +10963,22 @@ rb_str_force_encoding(VALUE str, VALUE enc)
 
 /*
  *  call-seq:
- *     str.b   -> str
+ *    b -> string
+ *
+ *  Returns a copy of +self+ with that has ASCII-8BIT encoding;
+ *  the contents (bytes) of +self+ are not modified:
+ *
+ *    s = "\x99"   # => "\x99"
+ *    s.encoding   # => #<Encoding:UTF-8>
+ *    t = s.b      # => "\x99"
+ *    t.encoding   # => #<Encoding:ASCII-8BIT>
+ *
+ *    s = "\u4095" # => "䂕"
```

Comment:
I think we should avoid using any non-US-ASCII characters in C source code.  None of the Ruby C files use non-US-ASCII characters:

```
ruby -e "ARGV.each{|f| p f if File.binread(f) =~ /[\x80-\xff]/}" *.c
```

Looks like there is an exception in the strscan extension, which should be fixed.

This issue affects other changes later in the commit.

Reply:
I've discovered before (then forgotten) that .c code must be ascii-only.
I've removed the return value for ```s = "*"``` here and a couple of other places.